test/contextualLifecycleRule.spec.ts

```diff
@@ -0,0 +1,113 @@
+import { getFailureMessage, Rule } from '../src/contextualLifecycleRule';
```

Comment:
I'd prefer to keep the previous specs. This is not very readable, it's hard to keep track what are the interfaces and the methods.

Reply:
No problem, I'll update ASAP.